### PR TITLE
Fix bug in caps lock light logic for Moonlander

### PR DESCRIPTION
**Pull request by Xaeroxe**
When the caps lock LED was first implemented it used layer 1's LED. This is no longer true, so this condition needs to be updated as caps lock is now using an LED for layer 3 and 6. Additionally this PR resolves a race condition happening between the LED layer code, and the caps lock LED code.

### CODE REVIEW

**Comment by Xaeroxe** [comment]
I'm happy to rebase this onto firmware_21 if that would be more helpful.

**Comment by Xaeroxe** [comment]
Seems like this merges cleanly into firmware21, so it is now targeting that.

**Comment by blueyed** [comment]
How does it relate to https://github.com/zsa/qmk_firmware/pull/335?

**Comment by Xaeroxe** [comment]
@blueyed #335 contains similar fixes to this one, however this PR resolves a race condition to make the lights on the keyboard flicker less and be more predictable. Also that PR introduces a compile time constant which allows you to use the caps lock LED exclusively for caps lock, rather than sharing it with the layer functionality.

**Comment by drashna** [comment]
Firmware 21 would probably be the better target. 

But as with #335, I think that it may be better to simple use LED 6 for caps lock exclusively, when it's enabled.

**Comment by Xaeroxe** [comment]
@drashna addressed in 2655f0d

**Comment by Xaeroxe** [comment]
Woops, missed one, got it in 772a9d1

File: keyboards/moonlander/moonlander.c [hunk 1]
```diff
@@ -129,18 +129,9 @@ void moonlander_led_task(void) {
         wait_ms(150);
     }
 #endif
-#ifdef CAPS_LOCK_STATUS
+#if !defined(MOONLANDER_USER_LEDS) && defined(CAPS_LOCK_STATUS)
```
Comment (drashna) [inline keyboards/moonlander/moonlander.c]:
```suggestion
#if !defined(MOONLANDER_USER_LEDS)
```

File: keyboards/moonlander/moonlander.c [hunk 2]
```diff
@@ -178,41 +169,58 @@ void keyboard_pre_init_kb(void) {
...
         default:
             break;
     }
 
+#ifdef CAPS_LOCK_STATUS
+    led_t led_state = host_keyboard_led_state();
+    if(led_state.caps_lock) {
+        LED_6 = true;
+    }
+#endif
```
Comment (drashna) [inline keyboards/moonlander/moonlander.c]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
And add: 
```c
#ifdef CAPS_LOCK_STATUS
bool led_update_kb(led_t led_state) {
    bool res = led_update_user(led_state);
    if(res) {
        ML_LED_6(led_state.caps_lock);
    }
    return res;
}
#endif
```

**Comment by Xaeroxe** [comment]
@drashna addressed in b86bfe6

File: keyboards/moonlander/moonlander.c [hunk 3]
```diff
@@ -178,41 +169,51 @@ void keyboard_pre_init_kb(void) {
...
         default:
             break;
     }
 
+    ML_LED_1(LED_1);
+    ML_LED_2(LED_2);
+    ML_LED_3(LED_3);
+    ML_LED_4(LED_4);
+    ML_LED_5(LED_5);
+    ML_LED_6(LED_6);
```
Comment (drashna) [inline keyboards/moonlander/moonlander.c:216]:
```suggestion
#if !defined(CAPS_LOCK_STATUS)
    ML_LED_6(LED_6);
#endif
```

**Pull request merged**